src/preload.js

```diff
@@ -22,6 +28,26 @@ function init() {
   window.MakerBridge = {
     getVersion,
   };
+  window.flashBoardFirmware = flashBoardFirmware;
+}
+
+function flashBoardFirmware(boardType) {
+  if (boardType === BOARD_TYPE.CLASSIC) {
+    let avrgirl = new Avrgirl({board: 'circuit-playground-classic'});
+    window.fetch('https://s3.amazonaws.com/downloads.code.org/maker/CircuitPlaygroundFirmata.ino.circuitplay32u4.hex')
+      .then(function(response) {
+        response.arrayBuffer().then(function(body) {
+          // Pass the response buffer to flash function to avoid filesystem error
+          avrgirl.flash(Buffer.from(body), (error) => {
+            if (error) {
+              console.error(error);
+            } else {
+              console.log('Firmware Updated');
+            }
+          });
+        });
+      });
+  }
```

Comment:
Since this is an asynchronous operation, it'd be nice if `flashBoardFirmware` returned a promise so whatever calls it can also know when flashing the board is complete.